### PR TITLE
Limit CurlFormatter usage for large requests

### PR DESCRIPTION
**Pull request by e-moe**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #189
| License       | Apache License 2.0

### CODE REVIEW

**Comment by csarrazi** [comment]
👍 Could you add a test for checking that the logic does work?

Thanks! :)

**Comment by e-moe** [comment]
@csarrazi , tests added

**Comment by e-moe** [comment]
something wrong with appveyor. is it possible to restart it?

File: tests/DataCollector/GuzzleCollectorTest.php [hunk 1]
```diff
@@ -74,5 +74,10 @@ public function testCollectCurlData()
             escapeshellarg('http://foo.bar')
         ), $calls[0]['curl']
         );
+
+        $client->post('http://foo.bar', ['body' => str_pad('', GuzzleCollector::MAX_BODY_SIZE + 1)]);
```
Comment (csarrazi) [inline tests/DataCollector/GuzzleCollectorTest.php]: You could simply initialise the collector with a small body size, but that's fine by me! :)

**Pull request merged**